Sources/Routing/RouteBuilder.swift

```diff
@@ -28,10 +30,56 @@ extension RouteBuilder {
     public func add(
         _ method: HTTP.Method,
         _ path: String ...,
-        _ value: @escaping RequestHandler
+        _ value: @escaping RouteHandler
     ) {
         let responder = Request.Handler { try value($0).makeResponse() }
         register(method: method, path: path, responder: responder)
     }
 
 }
+
+extension RouteBuilder {
+    public func socket(_ segments: String..., handler: @escaping WebSocketRouteHandler) {
+        register(method: .get, path: segments) { req in
+            return try req.upgradeToWebSocket {
+                try handler(req, $0)
+            }
+        }
+    }
+    
+    public func get(_ segments: String..., handler: @escaping RouteHandler) {
+        register(method: .get, path: segments) {
+            try handler($0).makeResponse()
+        }
+    }
+    
+    public func post(_ segments: String..., handler: @escaping RouteHandler) {
+        register(method: .post, path: segments) {
+            try handler($0).makeResponse()
+        }
+    }
+    
+    public func put(_ segments: String..., handler: @escaping RouteHandler) {
+        register(method: .put, path: segments) {
+            try handler($0).makeResponse()
+        }
+    }
+    
+    public func patch(_ segments: String..., handler: @escaping RouteHandler) {
+        register(method: .patch, path: segments) {
+            try handler($0).makeResponse()
+        }
+    }
+    
+    public func delete(_ segments: String..., handler: @escaping RouteHandler) {
+        register(method: .delete, path: segments) {
+            try handler($0).makeResponse()
+        }
+    }
+    
+    public func options(_ segments: String..., handler: @escaping RouteHandler) {
```

Comment:
We should have the generic `add` function here as well as an `all` function (should work by just adding `*` but i can't remember).